src/mscorlib/src/System/Guid.cs

```diff
@@ -1322,6 +1360,123 @@ unsafe private static int HexsToCharsHexOutput(char* guidChars, int a, int b)
+                if (dash) guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _b >> 8, _b);
+                if (dash) guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _c >> 8, _c);
+                if (dash) guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _d, _e);
+                if (dash) guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _f, _g);
+                offset += HexsToChars(guidChars + offset, _h, _i);
+                offset += HexsToChars(guidChars + offset, _j, _k);
+            }
+    }
+
+        // Returns whether the guid is successfully formatted as a span. 
+        public bool TryFormat(Span<char> destination, out int charsWritten, string format = null, IFormatProvider provider = null)
+        {
+            bool dash = true;
+            bool hex = false;
+            int offset = 0;
+            int guidSize;
+
+            if (format == null || format.Length == 0)
+                format = "D";
+
+            if (format.Length != 1)
+            {
+                // all acceptable format strings are of length 1
+                throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
+            }
+            char formatCh = format[0];
+            if (formatCh == 'D' || formatCh == 'd')
+            {
+                guidSize = 36;
+            }
+            else if (formatCh == 'N' || formatCh == 'n')
+            {
+                dash = false;
+                guidSize = 32;
+            }
+            else if (formatCh == 'B' || formatCh == 'b')
+            {
+                destination[offset++] = '{';
+                destination[37] = '}';
+                guidSize = 38;
+
+            }
+            else if (formatCh == 'P' || formatCh == 'p')
+            {
+                destination[offset++] = '(';
+                destination[37] = ')';
+                guidSize = 38;
+
+            }
+            else if (formatCh == 'X' || formatCh == 'x')
+            {
+                destination[offset++] = '{';
+                destination[67] = '}';
+                dash = false;
+                hex = true;
+                guidSize = 68;
+            }
+            else
+            {
+                throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
+            }
+
+            if (destination.Length < guidSize)
+            {
+                charsWritten = 0;
+                return false;
+            }
+            unsafe
+            {
+                fixed (char* guidChars = &destination.DangerousGetPinnableReference())
+                {
+                    FormatGuid(guidChars, ref offset, hex, dash);
+                }
+            }
+
+            charsWritten = guidSize;
+            return true;
+        }
+
         // IFormattable interface
         // We currently ignore provider
```

Comment:
When I proposed the TryFormat signature with the IFormatProvider argument, it was because Guid.ToString took one... I didn't realize at the time ToString completely ignores it.  @terrajobst, @weshaggard, what do you think, should we remove the IFormatProvider argument from TryFormat?